pkg/corerp/frontend/controller/applications/graph_util.go

```diff
@@ -497,21 +491,50 @@ func connectionsFromAPIData(resource generated.GenericResource) []*corerpv202310
 		data := corerpv20231001preview.ConnectionProperties{}
 		err := toStronglyTypedData(connection, &data)
 		if err == nil {
+			sourceID, ok := parseSource(to.String(resource.ID), to.String(data.Source))
+			if !ok {
+				continue
+			}
+
 			entries = append(entries, &corerpv20231001preview.ApplicationGraphConnection{
-				ID:        data.Source,
-				Direction: &dir,
+				ID:        to.Ptr(sourceID),
+				Direction: to.Ptr(dir),
 			})
 		}
 	}
 
 	// Produce a stable output
 	sort.Slice(entries, func(i, j int) bool {
-		return *entries[i].ID < *entries[j].ID
+		return to.String(entries[i].ID) < to.String(entries[j].ID)
 	})
 
 	return entries
 }
 
+func parseSource(parentID, source string) (string, bool) {
+	id, err := resources.Parse(source)
+	if err == nil && id.IsResource() {
+		return id.String(), true
+	}
+
+	sourceURL, err := url.Parse(source)
+	if err != nil {
+		return "", false
+	}
+
+	if sourceURL.Hostname() == "" || strings.Contains(sourceURL.Hostname(), ".") {
```

Comment:
What does this case mean?

Reply:
Updated the better way to parse only hostname case.